net/socket/Kconfig

```diff
@@ -43,7 +43,6 @@ config NET_SOLINGER
 	bool "SO_LINGER socket option"
 	default n
 	depends on NET_TCP_WRITE_BUFFERS || NET_UDP_WRITE_BUFFERS
-	select NET_TCP_NOTIFIER if NET_TCP
 	select NET_UDP_NOTIFIER if NET_UDP
```

Comment:
Why didn't NET_TCP_NOTIFIER need to be selected?
Could same be applied to NET_UDP_NOTIFIER later?

Reply:
@acassis ,

now all TCP data transmission is carried out by tcp_timer(), so we can delete the dependence on NET_TCP_NOTIFIER to avoid blocking for user thread.
UDP data transmission does not need to wait for the reply of the ACK confirmation as long as the data can be sent successfully, so we keep the logic of NET_UDP_NOTIFIER(txdrain) when udp close

Reply:
@acassis 

In this PR I also solved the indefinitely block issue in udp_close, please review this change 
https://github.com/apache/nuttx/pull/8129/commits/cb2671f34ee70013b13ecdafb382ebcadb96d551

net/udp: correct linger timeout

UDP linger timeout will be wrongly converted to UINT_MAX by _SO_TIMEOUT() when it is set to 0,

net/socket/socket.h:
```
|
| #  define _SO_TIMEOUT(t) ((t) ? (t) * MSEC_PER_DSEC : UINT_MAX)
```

net/udp/udp_close.c:
```
|
| if (_SO_GETOPT(conn->sconn.s_options, SO_LINGER))
|   {
|     timeout = _SO_TIMEOUT(conn->sconn.s_linger);
|   }
```

this change will correct this behavior, if the linger is set to 0, the timeout value should be 0

Signed-off-by: chao an <anchao@xiaomi.com>